prime_server/prime_server.hpp

```diff
@@ -70,7 +73,9 @@ template <class request_container_t, class request_info_t> class server_t {
            const std::string& interrupt_endpoint,
            bool log = false,
            size_t max_request_size = DEFAULT_MAX_REQUEST_SIZE,
-           uint32_t request_timeout = DEFAULT_REQUEST_TIMEOUT);
+           uint32_t request_timeout = DEFAULT_REQUEST_TIMEOUT,
+           const std::function<bool(const request_container_t&)>& health_check_matcher = {},
+           const std::string& health_check_response = {});
```

Comment:
these are the new options as describted, a matcher to identify when a request is a health check and the response to that when we see one